### PR TITLE
http: add CURLINFO_HTTP_VERSION and %{http_version}

### PR DESCRIPTION
**Pull request by gevaerts**
Adds access to the effectively used http version to both libcurl and curl.

Initial attempt, almost certainly not final yet.

### CODE REVIEW

**Comment by jay** [comment]
I like this idea, I have read of it a few times, as recently as [last month](https://curl.haxx.se/mail/lib-2016-04/0100.html). What do you think about using the existing CURL_HTTP_VERSION_X_X enumerations for getinfo, and then for the tool something like:
~~~c
verstring = ((ver == CURL_HTTP_VERSION_2_0) ? "2.0" :
             (ver == CURL_HTTP_VERSION_1_1) ? "1.1" :
             (ver == CURL_HTTP_VERSION_1_0) ? "1.0" : "0");
~~~

**Comment by gevaerts** [comment]
I've updated the patch to use CURL_HTTP_VERSION now.

**Comment by bagder** [comment]
:+1: to merge first thing after the pending release

**Comment by bagder** [comment]
Note that the travis-ci test failure is because the new symbol is not properly added to `docs/libcurl/symbols-in-versions`

**Comment by gevaerts** [comment]
I've added it to symbols-in-versions now.

**Comment by bagder** [comment]
Ganesh Nikam just posted a version of this patch on the mailing list:

  https://curl.haxx.se/mail/lib-2016-05/0131.html

He included a little example in his man page that could be merged into this patch to make it perfect I think!

File: src/tool_writeout.c [hunk 1]
```diff
@@ -278,6 +280,26 @@ void ourWriteOut(CURL *curl, struct OutStruct *outs, const char *writeinfo)
...
+                  char *version = "0";
+                  switch (longinfo) {
+                  case CURL_HTTP_VERSION_1_0:
+                    version = "1.0";
+                    break;
+                  case CURL_HTTP_VERSION_1_1:
+                    version = "1.1";
+                    break;
+                  case CURL_HTTP_VERSION_2_0:
+                    version = "2";
```
Comment (ganeshnikam18) [inline src/tool_writeout.c]: I think the string "2.0" would be better here instead of "2".

Comment (gevaerts) [inline src/tool_writeout.c]: The consensus seems to be that it's http 2, not 2.0. See e.g. https://http2.github.io/faq/#is-it-http20-or-http2

Comment (bagder) [inline src/tool_writeout.c]: The protocol is named HTTP/2 and there's no ".0" in the name or the version. I think we avoid a lot of problems by sticking to the official name.

**Pull request closed without merging**